### ISSUE TITLE
'module' object has no attribute 'LDAPBindError'

### ISSUE THREAD

**Issue by shanemgrey** [etianen/django-python3-ldap#55]
After provisioning a new VM with my app and dumping the DB from my dev db to it, I tried to login with the superuser account 'admin' that would have been setup by the provisioning script.  It failed with this error
 `'module' object has no attribute 'LDAPBindError'`

I realized my mistake was that the superuser would have the password from the dev db since I overwrote the new db completely.  Logging in with that password works.

My test VM **CANNOT** connect to the LDAP server right now, so I fully expected an error about failure to connect.  But from the looks of this, the exception handling isn't getting that far. 

The issue is that the error isn't saying the password was wrong, or that the connection can't be made.  It's saying the module doesn't have an attribute that the connection object should have to even tell which exception to throw.  And I don't know why.

My setup has the builtin ModelBackend being checked before LDAP.  So the correct superuser pw will avoid ever hitting the LDAP check, and therefore the error.

```
AUTHENTICATION_BACKENDS = [
    'django.contrib.auth.backends.ModelBackend',
    'django_python3_ldap.auth.LDAPBackend',
]
```

But with the wrong password, the check is passed on to the next backend, django_python3_ldap, which fails to throw the correct exception. 

```
Django Version:	1.10.4
Exception Type:	AttributeError
Exception Value:	
'module' object has no attribute 'LDAPBindError'
Exception Location:	/srv/api/lib/python3.4/site-packages/django_python3_ldap/ldap.py in connection, line 138
Python Executable:	/srv/api/bin/uwsgi
Python Version:	3.4.2
Python Path:	
['/srv/api/',
 '.',
 '',
 '/srv/api/lib/python3.4',
 '/srv/api/lib/python3.4/plat-x86_64-linux-gnu',
 '/srv/api/lib/python3.4/lib-dynload',
 '/usr/lib/python3.4',
 '/usr/lib/python3.4/plat-x86_64-linux-gnu',
 '/srv/api/lib/python3.4/site-packages']
```

**Comment by etianen** [comment]
Can you provide the full traceback of the exception?

**Comment by shanemgrey** [comment]
http://dpaste.com/3CXN01Q

**Comment by etianen** [comment]
Ah, I think that this is because the underlying ldap library broke it's
backwards compatibility. :(

Are you using v0.9.11 of django-python3-ldap? This was supposed to include
a version specifier that prevented the broken ldap library being pulled in
as a dependency.

**Comment by shanemgrey** [comment]
I've got v0.9.11 and ldap3 is 2.1.1  

This could be because I specified latest version for everything and named ldap3 explicitly in my requirements.  

I uninstalled both packages and installed django_python3_ldap only.  This installed ldap3 v1.4.0 and the authentication is now working as it should.

Thank you.

Is there a way to have multiple versions installed to deal with conflicting dependencies like this?  

I don't need ldap3 for anything else, but having pip respect the dependency version instead of using the one I installed explicitly would have avoided this issue.

**Comment by etianen** [comment]
Unfortunately, pip only supports one version of each python library. It's a bit of a limitation, certainly compared to NPM's hierarchal package management.

Glad it's fixed, anyway.

**Comment by shanemgrey** [comment]
Thanks for the quick solution to this.  And now I understand something new about pip!  

I did some looking into ways to avoid this in the future and found pipdeptree.  Pretty great utility for finding conflicts and enables other useful tricks like dumping just the parent packages to a requirements file.  Can be included in CI to fail the build if there are errors.

I appreciate this project.  It enabled me to integrate with my company auth system in a far easier way than any alternatives I had found.
